dev-tools/oxlog/src/lib.rs

```diff
@@ -0,0 +1,442 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! A tool to show oxide related log file paths
+//!
+//! All data is based off of reading the filesystem
+
+use anyhow::Context;
+use camino::Utf8PathBuf;
+use chrono::{DateTime, Utc};
+use std::collections::BTreeMap;
+use std::fs::{read_dir, DirEntry};
+use std::io;
+use uuid::Uuid;
+
+/// Return a UUID if the `DirEntry` contains a directory that parses into a UUID.
+fn get_uuid_dir(result: io::Result<DirEntry>) -> Option<Uuid> {
+    let Ok(entry) = result else {
+        return None;
+    };
+    let Ok(file_type) = entry.file_type() else {
+        return None;
+    };
+    if !file_type.is_dir() {
+        return None;
+    }
+    let file_name = entry.file_name();
+    let Some(s) = file_name.to_str() else {
+        return None;
+    };
+    if let Ok(uuid) = s.parse() {
+        Some(uuid)
+    } else {
+        None
+    }
+}
+
+#[derive(Debug)]
+pub struct Pools {
+    pub internal: Vec<Uuid>,
+    pub external: Vec<Uuid>,
+}
+
+impl Pools {
+    pub fn read() -> anyhow::Result<Pools> {
+        let internal = read_dir("/pool/int/")
```

Comment:
We don't actually read anything from the internal_pools, but I don't think it hurts to keep this type around for now. We may end up reading if we expand this tool to grab other files.

This was one of the first bits of code I wrote for this PR and I left it through all the iterations. Happy to change/remove if desired though.